src/main/java/com/example/books/mapper/ShoppingCartMapper.java

```diff
@@ -46,4 +38,26 @@ default ShoppingCartDto setCartItemsIds(
                         .collect(Collectors.toSet())
         );
     }
+
+    default User mapUserIdToUser(Long userId) {
+        if (userId == null) {
+            return new User();
+        }
+        User user = new User();
+        user.setId(userId);
+        return user;
+    }

```

Comment:
do you need this after you added  `@Mapping(source = "user.id", target = "userId")`?

Reply:
Yes, because @mapping maps id from user to userId and this method maps userId to user